README.md

```diff
@@ -143,6 +143,74 @@ camera.postDelayed(new Runnable() {
 
 ```
 
+### Error Handling
+
+#### Default Handler
+The default implementation will just throw all exceptions to prevent missing error handling.
+
+#### Basic Custom Handler
+You can implement custom error handling by overriding the ``onError`` listener.
+As soon as this method was overridden at least once (without calling the super method or re-throwing the exception),
+the default behavior will be disabled. So pay attention to not swallowing any exceptions.
+
+```java
+camera.addCameraListener(new CameraListener() {
+    @Override 
+    public void onError(CameraException exception) {
+        throw exception;
+    }
+});
+```
+
+#### Advanced Custom Handler
+Furthermore, you can distinguish different error types. E.g. some of them imply that the
+CameraView should be disabled or restarted while others may be ignored in some use cases.
+
+```java
+camera.addCameraListener(new CameraListener() {
+    @Override 
+    public void onError(CameraException exception) {
+        if (exception instanceof CameraUnavailableException) {
```

Comment:
This is a bit boring, and subclasses add no information other than their name. Can we just have `CameraException` to accept an `int` in its constructor representing the error type? For example

```java
public final static int TYPE_CONFIGURATION_FAILED = 1;
public final static int TYPE_PICTURE_FAILED = 2;
```

This way what we do is 

```java
switch (cameraException.getErrorType()) {
    case TYPE_CONFIGURATION_FAILED: ...
}
```

Reply:
I'd consider this a matter of taste. I prefer the different subclasses, especially because I like the hierarchical structure. When using an int, you can't group different capturing errors so easily etc.

Beside that, the current subclasses do not have further information yet, that's true. However, we keep the possibility to add such information only for the relevant cases. E.g. we could add an int as you proposed above, but only for the `CameraConfigurationFailedException` to indicate which configuration failed. This details aren't related to any of the other exception classes.